lib/fastlane/plugin/aws_s3/actions/aws_s3_action.rb

```diff
@@ -555,9 +559,9 @@ def self.available_options
                                        default_value: Actions.lane_context[SharedValues::IPA_OUTPUT_PATH]),
           FastlaneCore::ConfigItem.new(key: :xcarchive,
                                        env_name: "",
-                                       description: ".xcarchive file for the build ",
+                                       description: ".xcarchive file for the build - set to an empty string (`""`) to not use `Actions.lane_context[SharedValues::XCODEBUILD_ARCHIVE])` as default value",
                                        optional: true,
-                                       default_value: Actions.lane_context[SharedValues::XCODEBUILD_ARCHIVE]),
+                                       default_value: nil),
```

Comment:
What if we set the `default_value` to an empty string (`""`), so the `upload_xcarchive` won't be called for old users, and require:

- either a custom path to an archive
- either `nil`, in which case it would use the one from the `lane_context`

?

Reply:
yeah, good idea, seems good to me. can you please do the other change you propose in discussion? (about zipping the archive)
and then @joshdholtz could merge it if it's ok for him

Reply:
@claireduf, Done. Could you please review the changes so we make sure I didn't miss anything?